### PR TITLE
docs: Add docs for hide sensitive data feature

### PR DESCRIPTION
**Pull request by gagantrivedi**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/deployment/locally-api#pre-commit) to check linting
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?

## Changes

Add docs for `hide-sensitive data` feature

## How did you test this code?

N/A

### CODE REVIEW

**Review by novakzaballa — APPROVED** [review]
LGTM

File: docs/docs/advanced-use/hide-sensitive-data.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+- `flags[].id`
+- `flags[].feature.created_date`
+- `flags[].feature.description`
+- `flags[].feature.initial_value`
+- `flags[].feature.default_enabled`
+- `flags[].feature_segment`
+- `flags[].environment`
+- `flags[].identity`
+- `tratis[]`
```
Comment (matthewelwell) [inline docs/docs/advanced-use/hide-sensitive-data.md]:
```suggestion
- `traits[]`
```

File: docs/docs/advanced-use/hide-sensitive-data.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+            "name": "second_feature",
+            "type": "STANDARD"
+        },
+        "feature_state_value": null,
+        "enabled": false
+    }
+]
+```
+
+> **NOTE**: All fields mentioned are already excluded in the response generated by the Edge API.
```
Comment (matthewelwell) [inline docs/docs/advanced-use/hide-sensitive-data.md]: We should use the info callouts here, e.g.

```
:::info

All fields mentioned are already excluded in the response generated by the Edge API.

:::
```

File: docs/docs/advanced-use/hide-sensitive-data.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+                "type": "STANDARD"
+            },
+            "feature_state_value": null,
+            "enabled": false
+        }
+    ]
+}
+```
+
+> **NOTE**: Response generated by Edge API already excludes all the above mentioned fields apart from `traits`
```
Comment (matthewelwell) [inline docs/docs/advanced-use/hide-sensitive-data.md]: As above, let's use the info callouts.

**Review by dabeeeenster — APPROVED** [review]

**Pull request merged**